From bf0d336fc6bb69fd7732f497633252a259d9f596 Mon Sep 17 00:00:00 2001
From: "Wu, Qiming" <qi-ming.wu@intel.com>
Date: Tue, 3 Jul 2018 16:24:12 +0200
Subject: [PATCH] Merge pull request #29 in SW_UGW/linux from
 bugfix/DRVLIB_SW-804-fix-klocwork-error-report-in-vmb-module to master

* commit 'db00aab39862193b24b745e2f4d9295b96eb1fe5':
  DRVLIB_SW-804: fix klocwork report in vmb module
---
 arch/mips/lantiq/lantiq-vmb.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/mips/lantiq/lantiq-vmb.c b/arch/mips/lantiq/lantiq-vmb.c
index 63d92969245f..c9d3082e5b74 100644
--- a/arch/mips/lantiq/lantiq-vmb.c
+++ b/arch/mips/lantiq/lantiq-vmb.c
@@ -364,7 +364,7 @@ static int update_DB_from_DT(struct VMB_vpe_t *vt)
 		return ret;
 	}
 
-	strncpy(vt->name, name, sizeof(vt->name));
+	strlcpy(vt->name, name, sizeof(vt->name));
 	vt->cpu_status |= CPU_BOOTUP_DT;
 
 	pr_info("[%s]:[%d], cpuid = %d name = %s\n",
@@ -690,7 +690,7 @@ int8_t vmb_cpu_alloc(int8_t cpu, char *fw_name)
 			if ((vpet->cpu_status & CPU_BOOTUP_DT) == CPU_BOOTUP_DT)
 				vpet->cpu_status &= ~CPU_BOOTUP_DT;
 			else
-				strncpy(vpet->name, fw_name,
+				strlcpy(vpet->name, fw_name,
 					sizeof(vpet->name));
 #endif
 		} else {
@@ -719,7 +719,7 @@ int8_t vmb_cpu_alloc(int8_t cpu, char *fw_name)
 				if ((vpet[j].cpu_status & CPU_BOOTUP_DT) == CPU_BOOTUP_DT)
 					vpet[j].cpu_status &= ~CPU_BOOTUP_DT;
 				else
-					strncpy(vpet[j].name, fw_name,
+					strlcpy(vpet[j].name, fw_name,
 						sizeof(vpet[j].name));
 #endif
 				goto fin_alloc;
